roles/patch-vulns/tasks/main.yml

```diff
@@ -0,0 +1,36 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+- name: Remove known log4j files / folders
+  become: yes
+  become_user: root
+  file:
+    path: "{{ item }}"
+    state: absent
+  with_items:
+    - "{{ grid_home }}/tfa/{{ inventory_hostname_short }}/tfa_home/jlib/log4j-core-2.9.1.jar"
+    - "{{ grid_home }}/tfa/{{ inventory_hostname_short }}/tfa_home/jlib/tfa.war"
+    - "{{ grid_home }}/suptools/tfa/release/tfa_home/jlib/log4j-core-2.9.1.jar"
+    - "{{ grid_home }}/suptools/tfa/release/tfa_home/jlib/tfa.war"
+    - "{{ grid_home }}/.patch_storage"
+    - "{{ oracle_home }}/suptools/tfa/release/tfa_home/jlib/log4j-core-2.9.1.jar"
+    - "{{ oracle_home }}/suptools/tfa/release/tfa_home/jlib/tfa.war"
+    - "{{ oracle_home }}/md/property_graph/lib"
+    - "{{ oracle_home }}/md/property_graph/pgx"
+    - "{{ oracle_home }}/.patch_storage"
+    - "{{ swlib_path }}/V982063-01.zip"
```

Comment:
A few potential issues with deleting the base software install:
1. Makes future installs slower as the cached version needs to be re-copied from the GCS bucket
2. Due to 1 it's not particularly effective as the file exists on GCS anyway
3. It's 19c-specific - 18c,12.2 etc base releases are different of course

What do you think about leaving the base release in place, even if it contains vulnerable log4j components?

Reply:
Given the objective of this KR itself (rid log4j vulnerable jars/wars from BMS host), my thoughts are along the lines of:
* weighing the inconvenience of GCS /NFS copying again for a occasional re-invocation of `install-oracle.sh` for a failed run against
* ensuring the BMS host is wiped clean of any log4j  vulnerabilities
* I tend to lean towards removing the zip files.

What are your thoughts on this line of reasoning?

(Cleaning files existing on other non-BMS-host locations (like GCS or other NFS repositories) is a non-goal of this KR.)